src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileUploadsValidatorService.cs

```diff
@@ -195,13 +114,95 @@ private bool IsFileExisting(Guid releaseId, ReleaseFileTypes type, string name)
                 .Any(rf => String.Equals(rf.ReleaseFileReference.Filename, name, CurrentCultureIgnoreCase)
                 && rf.ReleaseFileReference.ReleaseFileType == type);
         }
+
+        private async Task<Either<ActionResult, bool>> ValidateDataFileNames(Guid releaseId, string dataFileName, string metaFileName)
+        {
+            if (string.Equals(dataFileName.ToLower(), metaFileName.ToLower(), OrdinalIgnoreCase))
+            {
+                return ValidationActionResult(DataAndMetadataFilesCannotHaveTheSameName);
+            }
+            
+            if (dataFileName.IndexOf(" ", Ordinal) > -1 || 
+                dataFileName.IndexOfAny(Path.GetInvalidFileNameChars()) > -1)
+            {
+                return ValidationActionResult(DataFilenameCannotContainSpacesOrSpecialCharacters);
+            }
+            
+            if (FileContainsSpacesOrSpecialChars(dataFileName))
+            {
+                return ValidationActionResult(FilenameCannotContainSpacesOrSpecialCharacters);
+            }
+            
+            if (metaFileName.IndexOf(" ", Ordinal) > -1 || 
+                metaFileName.IndexOfAny(Path.GetInvalidFileNameChars()) > -1)
+            {
+                return ValidationActionResult(MetaFilenameCannotContainSpacesOrSpecialCharacters);
+            }
+
+            if (FileContainsSpacesOrSpecialChars(metaFileName))
+            {
+                return ValidationActionResult(FilenameCannotContainSpacesOrSpecialCharacters);
+            }
+            
+            if (!metaFileName.ToLower().Contains(".meta."))
+            {
+                return ValidationActionResult(MetaFileIsIncorrectlyNamed);
+            }
+
+            if (IsFileExisting(releaseId, ReleaseFileTypes.Data, dataFileName))
+            {
+                return ValidationActionResult(CannotOverwriteDataFile);
+            }
+
+            if (IsFileExisting(releaseId, ReleaseFileTypes.Metadata, metaFileName))
+            {
+                return ValidationActionResult(CannotOverwriteMetadataFile);
+            }
+
+            return true;
+        }
+
+        private async Task<Either<ActionResult, bool>> ValidateDataFileSizes(long dataFileLength, long metaFileLength)
```

Comment:
If the Either's right is a `Unit` you don't need to return a `true`.

Reply:
done